mlrun/db/httpdb.py

```diff
@@ -1792,6 +1794,157 @@ def _validate_version_compatibility(server_version, client_version):
             )
             raise mlrun.errors.MLRunIncompatibleVersionError(message)
 
+    def create_or_patch(
+        self,
+        project: str,
+        endpoint_id: str,
+        model_endpoint: ModelEndpoint,
+        access_key: Optional[str] = None,
+    ):
+        """
+        Creates or updates a KV record with the given model_endpoint record
+
+        :param project: The name of the project
+        :param endpoint_id: The id of the endpoint
+        :param model_endpoint: An object representing the model endpoint
+        :param access_key: V3IO access key, when None, will be look for in environ
+        """
+        access_key = os.environ.get("V3IO_ACCESS_KEY", access_key)
```

Comment:
The precedence should be the opposite, if someone gave you access key, use it, only if it's None, try env